Userland/Applications/ClockSettings/ClockSettingsWidget.cpp

```diff
@@ -26,19 +40,82 @@ ClockSettingsWidget::ClockSettingsWidget()
     m_time_zone_combo_box->set_only_allow_values_from_model(true);
     m_time_zone_combo_box->set_model(*StringViewListModel::create(time_zones));
     m_time_zone_combo_box->set_text(m_time_zone);
+
+    auto time_zone_map_bitmap = Gfx::Bitmap::try_load_from_file("/res/graphics/map.png"sv).release_value_but_fixme_should_propagate_errors();
+    auto time_zone_rect = time_zone_map_bitmap->rect().shrunken(TIME_ZONE_MAP_NORTHERN_TRIM, 0, TIME_ZONE_MAP_SOUTHERN_TRIM, 0);
+    time_zone_map_bitmap = time_zone_map_bitmap->cropped(time_zone_rect).release_value_but_fixme_should_propagate_errors();
+
+    m_time_zone_map = *find_descendant_of_type_named<GUI::ImageWidget>("time_zone_map");
+    m_time_zone_map->set_bitmap(time_zone_map_bitmap);
+
+    auto time_zone_marker = Gfx::Bitmap::try_load_from_file("/res/icons/32x32/ladyball.png").release_value_but_fixme_should_propagate_errors();
+    m_time_zone_marker = time_zone_marker->scaled(0.75f, 0.75f).release_value_but_fixme_should_propagate_errors();
+
+    set_time_zone_location();
+}
+
+void ClockSettingsWidget::second_paint_event(GUI::PaintEvent& event)
+{
+    GUI::Widget::second_paint_event(event);
+
+    if (!m_time_zone_location.has_value())
+        return;
+
+    GUI::Painter painter(*this);
+    painter.add_clip_rect(event.rect());
+    painter.add_clip_rect(m_time_zone_map->relative_rect());
+
+    auto point = m_time_zone_location->to_type<int>();
+    point.translate_by(-m_time_zone_marker->width() / 2, -m_time_zone_marker->height() / 2);
+    point.translate_by(m_time_zone_map->x() + 8, m_time_zone_map->y() + 8);
+    painter.blit(point, *m_time_zone_marker, rect());
 }
 
 void ClockSettingsWidget::reset_default_values()
 {
     m_time_zone = "UTC"sv;
     m_time_zone_combo_box->set_text(m_time_zone);
+    m_time_zone_location.clear();
+
     set_time_zone();
+    update();
 }
 
 void ClockSettingsWidget::apply_settings()
 {
     m_time_zone = m_time_zone_combo_box->text();
+
+    set_time_zone_location();
     set_time_zone();
+    update();
+}
+
+void ClockSettingsWidget::set_time_zone_location()
+{
+    m_time_zone_location = compute_time_zone_location();
+}
+
+// https://en.wikipedia.org/wiki/Mercator_projection#Derivation
+Optional<Gfx::FloatPoint> ClockSettingsWidget::compute_time_zone_location() const
+{
+    auto location = TimeZone::get_time_zone_location(m_time_zone);
+    if (!location.has_value())
+        return {};
+
+    auto latitude = location->latitude.decimal_coordinate();
+    auto longitude = location->longitude.decimal_coordinate();
+
+    auto rect = m_time_zone_map->bitmap()->rect().to_type<float>();
+
+    latitude = logf(tanf(PI_OVER_4 + (latitude * PI_OVER_180 / 2.0f)));
+
+    auto mercadian_x = (longitude + 180.0f) * (rect.width() / 360.0f);
+    auto mercadian_y = (rect.height() / 2.0f) - (rect.width() * latitude / TAU);
+
+    mercadian_y -= TIME_ZONE_MAP_NORTHERN_TRIM / 2;
+    mercadian_y += TIME_ZONE_MAP_SOUTHERN_TRIM / 2;
```

Comment:
What is this supposed to do? I don't get why you use the halves here?

As far as I understand the mercadian_y is effectively a coordinate pretending the map spans the entire world. However you trim some dead space off so want to transform it to that new rect. But this doesn't do that correctly I think.

Maybe you want something like this:
`mercadian_y = mercadian_y * (full_image_height / trimmed_image_height) - TIME_ZONE_MAP_NORTHEN_TRIM`
(Convert coordinate to full image size then remove top trimmed part)

Although it is very possible I'm misunderstanding something here.
But the southern timezones you show in the video seem a little off to me.

Reply:
When computing `mercadian_y` above here, the formula halves the height. So this is applying the same 0.5 factor to these offsets.

If I don't trim the map at all (so those trim vars are 0), this is where the Antarctica pin gets dropped, which looks like the same position as the trimmed map:
![antarctica](https://user-images.githubusercontent.com/5600524/152274495-14911172-7245-4e4c-ba77-b5b752ecece9.png)

Reply:
Ah I must have messed up something while checking the math, seems fine then.